### PR TITLE
Update firewall.sh

### PR DESCRIPTION
**Pull request by jumpsmm7**
* Improve Regex to consider only Real IP addresses and Ranges

### CODE REVIEW

**Comment by jumpsmm7** [comment]
Don't merge yet

**Comment by jumpsmm7** [comment]
I think I covered all possible points of failure.

File: firewall.sh [hunk 1]
```diff
@@ -873,13 +873,12 @@ Whitelist_CDN() {
 		{
 			# Apple AS714 | Akamai AS12222 AS16625 | HighWinds AS33438 AS20446 | Fastly AS54113 | GitHub AS36459
 			for asn in AS714 AS12222 AS16625 AS33438 AS20446 AS54113 AS36459; do
-				curl -fsL --retry 3 --connect-timeout 3 "https://api.bgpview.io/asn/$asn/prefixes" | grep -oE '.{20}([0-9]{1,3}\.){3}[0-9]{1,3}\\/[0-9]{1,2}' | grep -vF "parent" | grep -oE '([0-9]{1,3}\.){3}[0-9]{1,3}\\/[0-9]{1,2}' | tr -d "\\" | awk -v asn="$asn" '{printf "add Skynet-Whitelist %s comment \"CDN-Whitelist: %s\"\n", $1, asn }' &
+				curl -fsL --retry 3 --connect-timeout 3 "https://asn.ipinfo.app/api/text/list/$asn" | awk -v asn="$asn" '/^(((25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])\.){3}(25[0-5]|(2[0-4]|1[0-9]|[1-9]|)[0-9])(\/(1?[0-9]|2?[0-9]|3?[0-2]))?)([[:space:]]|$)/{printf "add Skynet-Whitelist %s comment \"CDN-Whitelist: %s\"\n", $1, asn }'
```
Comment (dave14305) [inline firewall.sh:876]: Why are you changing the URL?

Comment (jumpsmm7) [reply firewall.sh:876]: There was an actual string error processing from that url. I manually ran each line of code to test it. The url had really bad parsing error. This url doesn't and gives same results.

Comment (jumpsmm7) [reply firewall.sh:876]: I see he has two other places where the site gets used, maybe for code consistency I should adjust those as well

Comment (dave14305) [reply firewall.sh:876]: It should really be a separate pull request. Don’t mix the regex changes with anything else. Harder to troubleshoot or revert.

Comment (jumpsmm7) [reply firewall.sh:877]: I can close this and reopen as one pull request

Comment (dave14305) [reply firewall.sh:877]: That wasn’t what I suggested. The ASN provider switch should not be in the same PR as the regex changes. Maybe @Adamm00 doesn’t want to change it, but only wants the regex tweaks.

My 2 cents…

Comment (jumpsmm7) [reply firewall.sh:877]: He is welcome to change the link back, I just thought he might want to find away to make it more compatible. However, I couldn't find away to make it more compatible. So here we are with the only solution I could come up with.

**Pull request closed without merging**